mvn.bzl

```diff
@@ -0,0 +1,41 @@
+def _mvn_package_impl(ctx):
+    inputs = []
+    inputs.extend(ctx.files.srcs)
+    inputs.append(ctx.file.pom_xml)
+
+    for input in inputs:
+        print(input.short_path)
+
+    output_jar = ctx.actions.declare_file("target/%s-%s.jar" % (ctx.attr.artifact_id, ctx.attr.version))
+    target_dir = ctx.actions.declare_directory("target")
+    outputs = [output_jar, target_dir]
+
+    # -Djar.finalName=custom-jar-name
+    ctx.actions.run_shell(
+        inputs = inputs,
+        outputs = outputs,
+        mnemonic = "MvnPackage",
+        progress_message = "Running 'mvn package' for %s" % output_jar.short_path,
+        command = "mvn package -DskipTests -DbazelOutputDir=%s -Pbazel" % (output_jar.dirname),
```

Comment:
We probably shouldn't expect `mvn` be in the system and on PATH, should it be lazily downloaded from https://maven.apache.org/download.cgi?

Reply:
it can be - this is just a quick prototype.

Reply:
Trying this rule when having the pom.xml not in the root package results in the mvn command failing as it does not find the pom.xml.
`mvn package -f %s -DskipTests -DbazelOutputDir=%s -Pbazel" % (ctx.file.pom_xml.dirname,output_jar.dirname)` worked for me.